crates/subspace-networking/src/utils/piece_announcement.rs

```diff
@@ -0,0 +1,155 @@
+use crate::node::Node;
+use crate::utils::multihash::ToMultihash;
+use backoff::future::retry;
+use backoff::ExponentialBackoff;
+use futures::StreamExt;
+use libp2p::core::multihash::Multihash;
+use std::collections::HashSet;
+use std::error::Error;
+use std::time::Duration;
+use subspace_core_primitives::PieceIndexHash;
+use tracing::{debug, trace, warn};
+
+const MAX_PEERS_TO_ACKNOWLEDGE: usize = 20; // Similar to Kademlia
+
+/// Defines initial duration between put_piece calls.
+const PUT_PIECE_INITIAL_INTERVAL: Duration = Duration::from_secs(1);
+/// Defines max duration between put_piece calls.
+const PUT_PIECE_MAX_INTERVAL: Duration = Duration::from_secs(30);
+
+fn default_backoff() -> ExponentialBackoff {
+    ExponentialBackoff {
+        initial_interval: PUT_PIECE_INITIAL_INTERVAL,
+        max_interval: PUT_PIECE_MAX_INTERVAL,
+        // Try until we get a valid piece
+        max_elapsed_time: None,
+        ..ExponentialBackoff::default()
+    }
+}
+
+pub async fn announce_piece<Pid: Into<PieceIndexHash>>(
```

Comment:
First of all, renaming of the file should be separated from significant changes into separate commits, please follow contributing guidelines, it hurts review process.
Second, I don't like that you have replaced separate functions that can do piece announcement with and without retries separately with one that just silently does some hardcoded retries without ability to not do them when not necessary.
Third, `Pid` is a very strange name for generic especially one that gets converted into `PieceIndexHash`, it should just be non-generic with `PieceIndexHash` as an argument. Calling `piece_index.hash()` is not that hard and makes code more readable.

We had 3 functions before, IMHO we need just two: with and without retries. Remember: this is a library, but you're trying to write application code in here with hardcoded logic. This is not appropriate for the context, it must be flexible.